theHarvester Enhancements and New Features

ibrahimsql opened the pull request:
# theHarvester Enhancements and New Features

## Changes Made

### Docker Improvements
- Fixed Docker configuration issues
- Optimized Dockerfile and docker-compose.yml files

### New Modules Added
- `whoisxml`: WHOIS XML API integration
- `venacus`: Venacus search engine integration
- `haveibeenpwned`: Data breach checking
- `leaklookup`: Leak database checking
- `securityscorecard`: Security score checking
- `builtwith`: Technology profile analysis
- `api_endpoints`: API endpoint scanning

### New Files

```
theHarvester/discovery/
├── api_endpoints.py
├── builtwith.py
├── haveibeenpwned.py
├── leaklookup.py
├── search_dehashed.py
├── securityscorecard.py
├── venacussearch.py
└── whoisxml.py
```


### Parser Added

```
theHarvester/parsers/
└── venacusparser.py
```

### Wordlist Directory
- Created `wordlists/` directory
- Added `wordlists/api_endpoints.txt`

## Testing
- All new modules have been tested
- Docker build and run tests completed
- API integrations verified

## Contributors
- @ibrahimsql

## Related Issue
- #991

L1ghtn1ng commented:
Can you do a git rebase please as there are merge conflicts that need sorting ☺️ major thanks for this☺️

ibrahimsql commented:
Sorry im still not used to using github :D im fixing it

L1ghtn1ng commented:
No worries, I created a git guide in the notes section of discover for Lee to help him out with git back in the day

L1ghtn1ng commented:
Recommend you take a look, will give you a boost in your git foo ☺️

ibrahimsql commented:
thanks a lot l1ghtn1ng, really appreciate it! 🙏 i’ll def check out the git guide, sounds super helpful. still getting used to git and rebasing 😅 but learning as i go. big thanks for the help!

L1ghtn1ng commented:
What I would also recommend is that you if you can avoid it is making more MR that are smaller than one giant one, makes things much easier to review

ibrahimsql commented:
There were some errors in the code (99) I fixed them

ibrahimsql commented:
Sorry for the trouble that was a skill issue on my part :( i`ve figured out where I went wrong  there should be no more merge conflicts now 🙃

L1ghtn1ng commented:
Both please

On 17 May 2025, at 13:29, yiğit ibrahim ***@***.***> wrote:

﻿

@ibrahimsql commented on this pull request.

________________________________

In theHarvester/lib/core.py<https://github.com/laramies/theHarvester/pull/1995#discussion_r2094111460>:

> +            'trello',
+            'twitter',


exactly which ones should i not remove?

—
Reply to this email directly, view it on GitHub<https://github.com/laramies/theHarvester/pull/1995#discussion_r2094111460>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AA3V2QXF5S5WJWG36QVXAT3264TTRAVCNFSM6AAAAAB5JV5Z7WVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNBYGI3DOOBQGU>.
You are receiving this because you commented.Message ID: ***@***.***>

L1ghtn1ng commented on theHarvester/lib/core.py:
```diff
@@ -188,32 +207,47 @@ def get_supportedengines() -> list[str | Any]:
+            'threatcrowd',
             'threatminer',
             'tomba',
+            'trello',
+            'twitter',
```
Both please

ibrahimsql commented on theHarvester/__main__.py:
```diff
@@ -407,6 +419,14 @@ async def store(
                         print(e)
 
+                        
+                elif engineitem == 'binaryedge':
```
removed

ibrahimsql commented:
All items that needed to be removed have been deleted.

L1ghtn1ng commented:
I have pushed new code and unfortunately it has created a merge conflict, I have moved to have the project deps to be in pyproject.toml only now to tidy things up so can you resync your branch please

L1ghtn1ng commented on README.md:
```diff
@@ -14,6 +14,8 @@ Passive modules:
 ----------------
 * anubis: Anubis-DB - https://github.com/jonluca/anubis
 
+* binaryedge: List of known subdomains (Requires an API key, see below.) - https://www.binaryedge.io
+
```
This needs removing as well

L1ghtn1ng commented on wordlists/api_endpoints.txt:
the wordlists directory is not needed as we have one already, so this file needs to just go in https://github.com/laramies/theHarvester/tree/master/theHarvester/data/wordlists and that will take care of it on install so it will always exist then

ibrahimsql replied:
deleted

L1ghtn1ng commented on README.md:
```diff
@@ -106,6 +108,7 @@ Modules that require an API key:
 Documentation to setup API keys can be found at - https://github.com/laramies/theHarvester/wiki/Installation#api-keys
 
 * bevigil - Free upto 50 queries. Pricing can be found here: https://bevigil.com/pricing/osint
+* binaryedge - $10/month
```
needs removing

L1ghtn1ng commented:
still need you to fix your merge conflict please :-)

ibrahimsql commented:
I can't see which merge conflicts
<img width="1312" alt="...." src="https://github.com/user-attachments/assets/68dce801-cb6c-476e-b3d1-e55cc33c6575" />

L1ghtn1ng commented:
So what you are seeing in the picture is the file in question that is the merge conflict that needs fixing

ibrahimsql commented:
I know but I can't see which ones

L1ghtn1ng commented:
requirements/dev.txt is your issue

ibrahimsql commented:
so how do i check for merge conflicts?

L1ghtn1ng commented:
Take a look at https://github.com/leebaird/discover/blob/main/notes/git.txt and you will want to look at syncing your fork section that will help you but recommend you read all of that file☺️

ibrahimsql commented:
i've added everything that is supported in fullhunt 
![...](https://github.com/user-attachments/assets/ba1d0654-56ef-4c3d-90f8-ef6ea106d015)

L1ghtn1ng commented:
Can you please run ruff for fixing the linting and formatting, thanks

ibrahimsql commented:
![20250520_020159](https://github.com/user-attachments/assets/4e1c795e-a690-452c-90d3-cf870403c010)

L1ghtn1ng commented on theHarvester/__main__.py:
```diff
@@ -1398,156 +1386,173 @@ async def handler(lst):
             # Define a default wordlist if none is specified
-            wordlist = args.wordlist if args.wordlist else "./wordlists/api_endpoints.txt"
-            
+            wordlist = args.wordlist if args.wordlist else './wordlists/api_endpoints.txt'
```
you should use the data dir to get api_endpoints.txt file which is where you need to put the file is pop it in here https://github.com/laramies/theHarvester/tree/master/theHarvester/data/wordlists

L1ghtn1ng commented on theHarvester/__main__.py:
```diff
@@ -1398,156 +1386,173 @@ async def handler(lst):
                 wordlist = temp_wordlist
-                print(f"Basic API wordlist created with {len(basic_endpoints)} endpoints.")
-            
-            print(f"\n[*] Starting API endpoint scanning with wordlist: {wordlist}")
```
depending on the above comment you will probably need to change this section

L1ghtn1ng commented:
I would say we are close to merging, just a few small tweaks and we should be good :-) then @leebaird can take everything for a proper spin ;-)

ibrahimsql commented:
Thank you for the review Hey @L1ghtn1ng, thanks a lot for the thorough review

i’ve gone ahead and implemented all your suggestions for the API endpoint scanning feature:
Wordlist paths now use DATA_DIR instead of being hardcoded
Cleaned up the duplicated code blocks
Quotes are now standardized throughout the module


> I would say we are close to merging, just a few small tweaks and we should be good :-) then @leebaird can take everything for a proper spin ;-)

Hopefully everything goes smoothly and we can get this feature into theHarvester as soon as possible
